extensions/jackson/src/main/java/io/jsonwebtoken/jackson/io/JacksonDeserializer.java

```diff
@@ -120,7 +120,8 @@ private MappedTypeDeserializer(Map<String, Class> claimTypeMap) {
         @Override
         public Object deserialize(JsonParser parser, DeserializationContext context) throws IOException {
             // check if the current claim key is mapped, if so traverse it's value
-            if (claimTypeMap != null && claimTypeMap.containsKey(parser.currentName())) {
+
+            if (claimTypeMap != null && parser.currentName() != null && claimTypeMap.containsKey(parser.currentName())) {
```

Comment:
Should we memoize `parser.currentName()` ?  It's called in this code 3 different times.